core/vm/tracer/call_stack_recorder.go

```diff
@@ -2,29 +2,53 @@ package tracer
 
 import (
 	"math/big"
+	"sync"
 
 	"github.com/ethereum/go-ethereum/common"
 	"github.com/ethereum/go-ethereum/core/tracing"
 )
 
-type CallStackRecorder struct {
+var (
+	callStackRecorderInit sync.Once
+	recorder              *callStackRecorder
```

Comment:
this is not good if multiple evms spin up at the same time, normal flow should be synchronized but when eth_simulate or a method debug is called, that would causes problems